isisd/isis_route.c

```diff
@@ -724,3 +730,38 @@ void isis_route_invalidate_table(struct isis_area *area,
 		UNSET_FLAG(rinfo->flag, ISIS_ROUTE_FLAG_ACTIVE);
 	}
 }
+
+void isis_route_switchover_nexthop(struct isis_area *area,
+				   struct route_table *table, afi_t afi,
+				   union g_addr *nexthop_addr,
+				   ifindex_t ifindex)
+{
+	struct isis_route_info *rinfo;
+	struct prefix_ipv6 *src_p;
+	struct route_node *rode;
+	struct prefix *prefix;
+
+	for (rode = route_top(table); rode; rode = srcdest_route_next(rode)) {
+		if (!rode->info)
+			continue;
+		rinfo = rode->info;
+
+		if (!rinfo->backup)
+			continue;
+
+		if (!nexthoplookup(rinfo->nexthops, afi, nexthop_addr, ifindex))
+			continue;
+
+		srcdest_rnode_prefixes(rode, (const struct prefix **)&prefix,
+				       (const struct prefix **)&src_p);
+
+		/* Switchover route. */
+		UNSET_FLAG(rinfo->flag, ISIS_ROUTE_FLAG_ZEBRA_SYNCED);
+		isis_route_update(area, prefix, src_p, rinfo->backup);
```

Comment:
will an existing SR SID be managed properly in this path? it looks like there's an expectation there that the 'route_info' carry state about the "before" SR info.

Reply:
I need to test a SR scenario